org.eclipse.xtext.ui.testing/src/org/eclipse/xtext/ui/testing/AbstractHoverTest.xtend

```diff
@@ -0,0 +1,120 @@
+/*******************************************************************************
+ * Copyright (c) 2018 itemis AG (http://www.itemis.eu) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.xtext.ui.testing
+
+import com.google.inject.Inject
+import org.eclipse.core.resources.IFile
+import org.eclipse.core.runtime.NullProgressMonitor
+import org.eclipse.jface.internal.text.html.BrowserInformationControlInput
+import org.eclipse.jface.text.IRegion
+import org.eclipse.jface.text.ITextHoverExtension2
+import org.eclipse.jface.text.Region
+import org.eclipse.xtext.resource.FileExtensionProvider
+import org.eclipse.xtext.ui.XtextProjectHelper
+import org.eclipse.xtext.ui.editor.XtextEditor
+import org.eclipse.xtext.ui.editor.XtextEditorInfo
+import org.eclipse.xtext.ui.editor.hover.IEObjectHover
+import org.eclipse.xtext.ui.refactoring.ui.SyncUtil
+import org.eclipse.xtext.ui.testing.util.IResourcesSetupUtil
+
+import static extension org.eclipse.xtext.ui.testing.util.IResourcesSetupUtil.addNature
+
+/**
+ * @author miklossy - Initial contribution and API
+ * 
+ * @since 2.16
+ */
+class AbstractHoverTest extends AbstractEditorTest {
+
+	@Inject protected IEObjectHover eObjectHover
+	@Inject protected XtextEditorInfo editorInfo
+
+	@Inject protected extension SyncUtil
+	@Inject protected extension FileExtensionProvider
+
+	override protected getEditorId() {
+		editorInfo.getEditorId
+	}
+
+	/**
+	 * Test that the expected text appears in a popup window while hovering over a certain text in a given DSL text.
+	 * 
+	 * @param it The initial DSL text.
+	 * @param hoverText The text hovering over.
+	 * @param hoverContent The text that are expected to be part of the hover popup.
+	 */
+	def void hasHoverOver(CharSequence it, String hoverText, String hoverContent) {
+		hasHoverOver(new Region(toString.indexOf(hoverText), hoverText.length), hoverContent)
+	}
+
+	/**
+	 * Test that the expected text appears in a popup window while hovering over a certain region in a given DSL text.
+	 * 
+	 * @param it The initial DSL text.
+	 * @param hoverRegion The region hovering over.
+	 * @param hoverContent The text that are expected to be part of the hover popup.
+	 */
+	def void hasHoverOver(CharSequence it, IRegion hoverRegion, String hoverContent) {
+		// given
+		dslFile.
+		// when
+		hoveringOver(hoverRegion).
+		// then
+		hoverPopupHasContent(hoverContent)
+	}
+
+	protected def IFile dslFile(CharSequence content) {
+		val file = IResourcesSetupUtil.createFile(projectName, fileName, fileExtension, content.toString)
+
+		/*
+		 * TODO: find a better (with good performance) solution
```

Comment:
What's wrong with this solution?

Reply:
See also https://github.com/eclipse/xtext-eclipse/issues/750